Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/APINavPage.xaml.cs

```diff
@@ -0,0 +1,160 @@
+
+            xTreeView.Tree.TreeNodesFilterByField = new Tuple<string, string>(nameof(ApplicationAPIModel.TargetApplicationKey) + "." + nameof(ApplicationAPIModel.TargetApplicationKey.ItemName), mContext.BusinessFlow.CurrentActivity.TargetApplication);
+            xTreeView.Tree.FilterType = UCTreeView.eFilteroperationType.Equals;
+            TreeViewItem r = xTreeView.Tree.AddItem(itemTypeRootNode);
+
+            r.IsExpanded = true;
+
+            itemTypeRootNode.SetTools(xTreeView);
+            xTreeView.SetTopToolBarTools(saveAllHandler, addHandler);
+
+            xTreeView.Tree.ItemSelected += MainTreeView_ItemSelected;
+            SetElementsGridView();
+
+            //if (treeItemDoubleClickHandler != null)
+            //{
+            //    xTreeView.Tree.ItemDoubleClick += treeItemDoubleClickHandler;
+            //}
+        }
+
+        private void MContext_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            mContext.Activity.PropertyChanged -= Activity_PropertyChanged;
+            mContext.Activity.PropertyChanged += Activity_PropertyChanged;
+            if (e.PropertyName is nameof(mContext.BusinessFlow) || e.PropertyName is nameof(mContext.Activity))
+            {
+                UpdateAPITree();
+            }
+        }
+
+        private void UpdateAPITree()
+        {
+            if (mContext.BusinessFlow.CurrentActivity == null)
+                mContext.BusinessFlow.CurrentActivity = mContext.BusinessFlow.Activities[0];
+
+            xTreeView.Tree.TreeNodesFilterByField = new Tuple<string, string>(nameof(ApplicationAPIModel.TargetApplicationKey) + "." + nameof(ApplicationAPIModel.TargetApplicationKey.ItemName), mContext.BusinessFlow.CurrentActivity.TargetApplication);
+            xTreeView.Tree.FilterType = UCTreeView.eFilteroperationType.Equals;
+            xTreeView.Tree.RefresTreeNodeChildrens(mItemTypeRootNode);
+        }
+
+        private void Activity_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e.PropertyName == nameof(mContext.Activity.TargetApplication))
+            {
+                UpdateAPITree();
+            }
+        }
+
+        private void CurrentActivity_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            //xTreeView.Tree.TreeNodesFilterByField = new Tuple<string, string>(nameof(ApplicationPOMModel.TargetApplicationKey) + "." + nameof(ApplicationPOMModel.TargetApplicationKey.ItemName), mContext.BusinessFlow.CurrentActivity.TargetApplication);
+            //xTreeView.Tree.FilterType = UCTreeView.eFilteroperationType.Equals;
+        }
+
+        private void MainTreeView_ItemSelected(object sender, EventArgs e)
+        {
+            //GridLength POMDetailsRegionHeight = new GridLength(400, GridUnitType.Star);
+            //GridLength unloadedPOMDetailsHeight = new GridLength(0);
```

Comment:
please delete all commented out code if not needed

Reply:
Done !